blogService/src/main/java/blogService/service/UserService.java

```diff
@@ -36,12 +72,65 @@ public UserDto getUser(Long userId) {
 
     public String deleteUser(Long userId, String password) {
         Optional<User> optional = userRepository.findById(userId);
-        if (optional.isPresent()) {
-            if (password.hashCode() == userRepository.findById(userId).get().getPassHash()) {
-                userRepository.deleteById(userId);
-                return "The user has been deleted.";
-            }
+        if (optional.isPresent() && password.hashCode() == userRepository.findById(userId).get().getPassHash()) {
```

Comment:
zauważ że już wywołałaś `userRepository.findById(userId)` a wartość została przypisana do `optional`. Proponuję w takim razie taki zapis:
`optional.isPresent() && password.hashCode() == optional.get().getPassHash()` 
nie dość że unikasz ponownego wywołania tej metody, to jeszcze pozbędziesz się warninga z Intellij (wywołanie `get` bez wcześniejszego `isPresent`)

Ponadto, nazwa zmiennej 'optional' jest mało opisowa, lepiej byłoby chociaż `userOptional`